specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,305 @@
+                            // whether or not to allow the protocol to launch.
+                        }
+                    return S_OK;
+                })
+                .Get(),
+            &m_launchingExternalProtocolToken));
+    }
+    auto webView4 = m_webview.try_query<ICoreWebView2_4>();
+    if (webView4)
+    {
+        // Note that FrameCreated will only ever be raised for top level iframes.
+        // However, any launching external protocol requests from nested iframes 
+        // will be raised from the top level frame.
+        CHECK_FAILURE(webView4->add_FrameCreated(
+            Callback<ICoreWebView2FrameCreatedEventHandler>(
+                [this](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args) -> HRESULT
+                {
+                    wil::com_ptr<ICoreWebView2Frame> webViewFrame;
+                    CHECK_FAILURE(args->get_Frame(&webViewFrame));
+
+                    auto webViewFrame3 = webViewFrame.try_query<ICoreWebView2Frame3>();
+                    if (webViewFrame3)
+                    {
+                        CHECK_FAILURE(webViewFrame3->add_LaunchingExternalProtocol(
+                        Callback<ICoreWebView2FrameLaunchingExternalProtocolEventHandler>(
+                            [this](
+                                ICoreWebView2Frame* sender,
+                                ICoreWebView2LaunchingExternalProtocolEventArgs* args) {
+                                    wil::unique_cotaskmem_string uri;
+                                    CHECK_FAILURE(args->get_Uri(&uri));
+                                    if (wcsicmp(uri.get(), L"calculator://") == 0)
+                                    {
+                                        // If this matches our desired protocol, then suppress the
+                                        // popup dialog.
+                                        CHECK_FAILURE(args->put_Handled(TRUE));
+                                    }
+                                    else
+                                    {
+                                        // Otherwise revoke permissions previously granted to this protocol 
+                                        // from this origin as well as disable the always open checkbox.
+                                        CHECK_FAILURE(args->put_DisableAlwaysOpenCheckbox(TRUE));
+                                    }
+                                    return S_OK;
+                            })
+                            .Get(),
+                        &m_frameLaunchingExternalProtocolToken));
+                    }
+                    return S_OK;
+            }).Get(),
+            &m_frameCreatedToken));
+    }
+}
+    
+```
+
+## .NET and WinRT
+
+```c #
+private WebView2 webView;
+void RegisterLaunchingExternalProtocolHandler() 
+{
+    // Safeguarding the handler when unsupported runtime is used.
```

Comment:
We've decided to exclude the try/catch NotImplementedException from the sample code in the reference docs & API docs.